src/Settings.cs

```diff
@@ -28,15 +26,60 @@ internal class HouseLightsSettings : JsonModSettings
         [Name("Colorless lights")]
         [Description("If set to yes, lights will cast a more white light. If set to no, they will cast light with the default color.")]
         public bool whiteLights = false;
-    }
 
+        [Name("Stove genrator")]
+        [Description("If set to yes, stoves are working as power generator and needed to be at certian temperature to get lights. Outside lights will be disabled. If not, lights work regardless of stove status and outside light can be used.")]
+        public bool stoveGenerator = false;
+
+        [Name("Generator min temp")]
+        [Description("Minimal stove temperature for electricity to flow. Recommended 15")]
+        [Slider(0f, 80f)]
+        public float stoveGeneratorMinTemp = 15f;
+
+        [Name("Generator temp")]
+        [Description("Optimal stove temperature for electricity to flow; if temperature is lower than this, it will reduce light intesivity. Recommended 50+")]
+        [Slider(0f, 80f)]
+        public float stoveGeneratorTemp = 50f;
+
+
+
+        protected override void OnChange(FieldInfo field, object oldValue, object newValue)
+        {
+            RefreshFields();
+        }
+
+        internal void RefreshFields()
+        {
+            if (stoveGenerator)
+            {
+                SetFieldVisible(nameof(stoveGeneratorTemp), true);
+                SetFieldVisible(nameof(stoveGeneratorMinTemp), true);
+            }
+            else
+            {
+                SetFieldVisible(nameof(stoveGeneratorTemp), false);
+                SetFieldVisible(nameof(stoveGeneratorMinTemp), false);
+            }
+            if (stoveGeneratorMinTemp >= stoveGeneratorTemp)
+            {
+                stoveGeneratorMinTemp = stoveGeneratorTemp - 1;
+                RefreshGUI();
```

Comment:
This is weird a bit.
It works on slider but not always on numeric representation. If you try to increase (by dragging slider) the min value to be above optimal one, slider will stay at maximum allowed value, but numeric representation goes as it pleases.
Not sure if this is some ModSettings issue. Value saved is OK.